fuse/eval/metrics/sequence_gen/metrics_seq_gen_common.py

```diff
@@ -66,6 +66,10 @@ def _perplexity_update(
 
     assert len(preds.shape) == 3, f"Error: expected num dims is 3, got shape {preds.shape}"
     assert len(target.shape) == 2, f"Error: expected num dims is 2, got shape {target.shape}"
+    # to save GPU memory
+    preds = preds.detach()
```

Comment:
Do you prefer to keep them on GPU?
would moving them to CPU make sense? (to free GPU mem)

detach() good but probably only helpful to avoid gradients calculation (and activations storage)